core/wasm/src/alloc.rs

```diff
@@ -0,0 +1,30 @@
+use core::alloc::Layout;
+use core_alloc::alloc;
+
+/// Exposes [`alloc::alloc`] to be called via FFI
+///
+/// # Safety
+///
+/// Check [`alloc::alloc`] documentation
+#[no_mangle]
+pub unsafe extern "C" fn _iroha_wasm_alloc(size: usize, align: usize) -> *mut u8 {
+    if let Ok(layout) = Layout::from_size_align(size, align) {
+        return alloc::alloc(layout);
+    }
+
+    core::ptr::null_mut()
+}
+
+/// Exposes [`alloc::dealloc`] to be called via FFI
+///
+/// # Safety
+///
+/// Check [`alloc::dealloc`] documentation
+#[no_mangle]
+pub unsafe extern "C" fn _iroha_wasm_dealloc(ptr: *mut u8, size: usize, align: usize) {
+    if let Ok(layout) = Layout::from_size_align(size, align) {
+        alloc::dealloc(ptr, layout);
+    } else {
+        /* NOTE: Leak memory */
```

Comment:
And this is alright, since the smartcontract is loaded into a fresh VM and everything gets deallocated anyway?

Reply:
ok, yes every module instance is sandboxed. But the main reason is that there is no way to communicate the error to the caller and panicking over FFI is UB. Also, this is a developer error, It's unlikely that the user would call such low level APIs